core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -713,6 +742,22 @@ class CivilizationInfo {
         addNotification("[${otherCiv.civName}] gave us a [${militaryUnit.name}] as gift near [${city.name}]!", locations, otherCiv.civName, militaryUnit.name)
     }
 
+    /** Gain a random great person from a random city state */
+    private fun gainGreatPersonFromCityState() {
+        val givingCityState = getKnownCivs().filter { it.isCityState() && it.getAllyCiv() == civName}.random()
+        val giftedUnit = getGreatPeople().random()
```

Comment:
This should be givingCityState.getGreatPeople() so you don't get your own unique GP from a city state